tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py

```diff
@@ -270,7 +307,35 @@ def get_command_line_arguments(argv):
         parameters.append((arg_name, arg_value))
     return parameters
 
-def parse_input_parameters(parameter_string, input_info):
+
+def get_input_output_names(nodes):
+    return [node.get_friendly_name() for node in nodes] # get_friendly_name() or get_name() ?
+
+
+def get_tensor_shapes_map(tensor_shape_string, input_names):
+    # Parse parameter string like "input0[shape1][shape2],input1[shape1]" or "[shape1][shape2]" (applied to all inputs)
+    return_value = {}
+    if tensor_shape_string:
+        tensor_shape_string += ','
+        matches = re.findall(r'(.*?\[.*?\]),', tensor_shape_string)
+        if matches:
+            for match in matches:
+                input_name = match[:match.find('[')]
+                shapes = re.findall(r'\[(.*?)\]', match[len(input_name):])
+                if input_name:
+                    return_value[input_name] = list(parse_partial_shape(shape_str) for shape_str in shapes)
+                elif len(input_names) == 1:
+                    return_value[input_names[0]] = list(parse_partial_shape(shape_str) for shape_str in shapes)
```

Comment:
What if `input_name`  is not equal to `input_names[0]` due to user mistake?

Reply:
In this case `function` has only one input named `input_names[0]` and user doesn't provide any input names.

Reply:
@AlexeyLebedev1 Who will forbid him to do this?

Reply:
Updated. Current logic is that user can pass `-data_shape` without name mapping even in case multi input network. `-data_shape shape1, shape2` will be parsed as `{input1: shape1, input2: shape2`} in case two inputs and `{input1: shape1, shape2}` in case one input for example.